### PR TITLE
Add issue_template

### PR DESCRIPTION
**Pull request by sethvargo**
This adds a new issues template modeled after the one we've been using for Vagrant. It mirrors a lot of the information already contained in the CONTRIBUTING.md, but provides a consistent template for new issues to follow as well as some examples.

This text will pre-fill when someone goes to create a new issue.

/cc @phinze @catsby

### CODE REVIEW

**Comment by catsby** [comment]
LGTM

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+issue tracker reserved for bug reports and feature requests. For general usage
+questions, please see: https://www.terraform.io/community.html.
+
+### Terraform Version
+Run `terraform -v` to show the version. If you are not running the latest
+version of Terraform, please upgrade because your issue may have already been
+fixed.
+
+### Affected Resource(s)
+Please list the resources as a list, for example:
```
Comment (jen20) [inline .github/ISSUE_TEMPLATE.md]: Maybe also something like: "if the issue appears to affect multiple resources and may be an issue with the core of Terraform, please also mention this".

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+### Debug Output
+Please provider a link to a GitHub Gist containing the complete debug output:
+https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the
+debug output in the issue; just paste a link to the Gist.
+
+### Panic Output
+If Terraform produced a panic, please provide a link to a GitHub Gist containing
+the output of the `panic.log`.
```
Comment (jen20) [inline .github/ISSUE_TEMPLATE.md]: I think the file is called `crash.log` from Terraform.

File: .github/ISSUE_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+### Expected Behavior
+What should have happened?
+
+### Actual Behavior
+What actually happened?
+
+### Steps to Reproduce
```
Comment (jen20) [inline .github/ISSUE_TEMPLATE.md]: Maybe worth a "anything unusual about your accounts" here - e.g. EC2 classic, custom OpenStack etc?

**Comment by sethvargo** [comment]
@jen20 added those things

**Comment by radeksimko** [comment]
I kind of like the Homebrew's checklist-style: https://github.com/Homebrew/homebrew-core/blob/master/.github/ISSUE_TEMPLATE.md

I wonder if there's any way we could turn this into a checklist too.

I'm thinking it may feel more natural for people reporting bugs to tick boxes and fill in gaps in the text, instead of removing it and/or answering questions.

e.g. 

 - [ ] I'm running the latest Terraform version (confirmed via `terraform -v`)
 - [ ] I'm reporting crash
   - [ ] I did attach link to a gist containing `crash.log`

**Comment by sethvargo** [comment]
Hi @radeksimko 

The problem I have with Homebrew's template is that those checkboxes only show up _after_ you create the issue, and they aren't versioned or anything. As a result, anyone with permissions can check/uncheck, and when I get an email from GitHub, it might not contain the right "checkboxes". For non-technical users, they might not be familiar with the markdown syntax for `[ ]` and I don't find the checkboxes helpful to be quite honest.

**Comment by radeksimko** [comment]
@sethvargo Thanks for the explanation, these reasons make sense.

Sometimes I see things only from my POV - i.e. someone who knows markdown quite well and almost doesn't see the difference between raw and rendered markdown. :smiley:

**Pull request merged**